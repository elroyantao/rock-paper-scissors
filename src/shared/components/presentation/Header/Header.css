.Header {
  margin-bottom: 30px;
}

.Header-tabs {
  display: flex;
}

.Header-tab {
  flex: 1;
  height: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
  border: 1px solid gray;
  cursor: pointer;
}

.Header-tab--active {
  composes: Header-tab;
  color: #59d482;
  border-color: #59d482;
  font-weight: bold;
}
