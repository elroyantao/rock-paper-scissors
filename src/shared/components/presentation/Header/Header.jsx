import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { changeLevel } from '../../../actions/levelAction'
import styles from './Header.css'

@connect((state) => ({
  level: state.level.name
}), { changeLevel })
export default class Header extends Component {
  static propTypes = {
    level: PropTypes.string,
    changeLevel: PropTypes.func
  }

  onChangeLevel(levelTo) {
    const { level, changeLevel } = this.props
    if (level === levelTo) return
    changeLevel(levelTo)
  }

  render() {
    const { level } = this.props
    return (
      <div className={styles.Header}>
        <div className={styles['Header-tabs']}>
          <div
            className={styles[`Header-tab${level === 'classic' ? '--active' : ''}`]}
            onClick={() => this.onChangeLevel('classic')}
          >
            Classic
          </div>
          <div
            className={styles[`Header-tab${level === 'advanced' ? '--active' : ''}`]}
            onClick={() => this.onChangeLevel('advanced')}
          >
            Advanced
          </div>
        </div>
      </div>
    )
  }
}
