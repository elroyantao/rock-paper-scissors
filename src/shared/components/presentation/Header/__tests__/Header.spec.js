import testComponentHelper from '../../../../lib/test-helper'
import Header from '../Header'


describe('<Header /> ', () => {
  const initialState = {
    level: 'classic',
    changeLevel: jest.fn()
  }
  const renderComponent = testComponentHelper(Header.WrappedComponent)
  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(initialState).getTree())
        .toMatchSnapshot()
    })
    it('in advanced state', () => {
      expect(renderComponent({
        ...initialState,
        level: 'advanced'
      }).getTree())
        .toMatchSnapshot()
    })
  })

  describe('@evets', () => {
    beforeEach(() => jest.clearAllMocks())
    it('on click of classic tab when level is classic it should not call changeLevel', () => {
      const { wrapper, instance } = renderComponent(initialState)
      wrapper.find('.Header-tabs > div').first().simulate('click')
      expect(instance.props.changeLevel).not.toHaveBeenCalled()
    })
    it('on click of classic tab when level is advanced it should call changeLevel', () => {
      const { wrapper, instance } = renderComponent({
        ...initialState,
        level: 'advanced'
      })
      wrapper.find('.Header-tabs > div').first().simulate('click')
      expect(instance.props.changeLevel).toHaveBeenCalledTimes(1)
      expect(instance.props.changeLevel).toHaveBeenLastCalledWith('classic')
    })
    it('on click of advanced tab when level is advanced it should not call changeLevel', () => {
      const { wrapper, instance } = renderComponent({
        ...initialState,
        level: 'advanced'
      })
      wrapper.find('.Header-tabs > div').last().simulate('click')
      expect(instance.props.changeLevel).not.toHaveBeenCalled()
    })
    it('on click of advanced tab when level is classic it should call changeLevel', () => {
      const { wrapper, instance } = renderComponent(initialState)
      wrapper.find('.Header-tabs > div').last().simulate('click')
      expect(instance.props.changeLevel).toHaveBeenCalledTimes(1)
      expect(instance.props.changeLevel).toHaveBeenLastCalledWith('advanced')
    })
  })
})
