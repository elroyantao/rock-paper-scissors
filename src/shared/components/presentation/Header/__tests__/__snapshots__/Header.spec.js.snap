// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header />  @renders in advanced state 1`] = `
<div
  className="Header"
>
  <div
    className="Header-tabs"
  >
    <div
      className="Header-tab"
      onClick={[Function]}
    >
      Classic
    </div>
    <div
      className="Header-tab--active"
      onClick={[Function]}
    >
      Advanced
    </div>
  </div>
</div>
`;

exports[`<Header />  @renders in default state 1`] = `
<div
  className="Header"
>
  <div
    className="Header-tabs"
  >
    <div
      className="Header-tab--active"
      onClick={[Function]}
    >
      Classic
    </div>
    <div
      className="Header-tab"
      onClick={[Function]}
    >
      Advanced
    </div>
  </div>
</div>
`;
